californium-core/src/main/java/org/eclipse/californium/core/network/Endpoint.java

```diff
@@ -82,16 +82,17 @@
 	boolean isStarted();
 
 	/**
-	 * Sets the executor for this endpoint and all its components.
+	 * Sets executors for this endpoint and all its components.
 	 *
-	 * The executor is not managed by the endpoint, it must be shutdown
+	 * Executors are not managed by the endpoint, it must be shutdown
 	 * externally, if the resource should be freed.
 	 *
-	 * @param executor the new executor
+	 * @param mainExecutor executors used for main tasks
+	 * @param secondaryExecutor intended to be used for rare executing timers (e.g. cleanup tasks). 
```

Comment:
Please document the `null` expectations.

Reply:
That's the same comment as above, but a bit better as it ends with a `.` instead of a `!` :grin: